src/curl.c

```diff
@@ -313,6 +314,16 @@ double swupd_curl_query_content_size(char *url)
 		return -1;
 	}
 
+	/* if the file doesn't exist in the remote server return a size of 0
+	 * otherwise a value of 162 will be returned for a non existent file
```

Comment:
Why 162?

Reply:
Because when querying the content server for a resource, and the resource is not found, the server will respond with a "404 Not Found" page which has a size itself. In the case of the default content server, it is serving the content using an nginx server, and the size of the 404 page from nginx is 162 bytes.
```
$ curl -I "https://cdn.download.clearlinux.org/update/29750/files/fake"
HTTP/1.1 200 Connection established

HTTP/2 404
server: nginx
content-type: text/html
content-length: 162
vary: Accept-Encoding
date: Mon, 24 Jun 2019 22:25:33 GMT
```
 
This value however can be different with other servers:
```
$ curl -I "http://somefakepage.org"
HTTP/1.1 404 Not Found
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/html; charset=utf-8
Content-Length: 662
Via: 1.1 jfintpr02
Proxy-Connection: Keep-Alive
Connection: Keep-Alive
Date: Mon, 24 Jun 2019 22:32:27 GMT
```

I will update the comment so it is better explained

Reply:
ok, now it makes sense